src/test/java/fi/metatavu/metaform/server/test/functional/builder/impl/ReplyTestBuilderResource.kt

```diff
@@ -0,0 +1,414 @@
+        return api.listReplies(metaformId, null, null, null, null, null, null, null, null, null)
+    }
+
+    /**
+     * Creates new reply for TBNC form
+     *
+     * @param metaformId metaform id
+     * @param text      text value
+     * @param bool      boolean value
+     * @param number    number value
+     * @param checklist checklist value
+     * @return reply
+     */
+    @Throws(IOException::class)
+    fun createTBNCReply(metaformId: UUID, text: String, bool: Boolean?, number: Double, checklist: Array<String?>): Reply {
+        val replyData: MutableMap<String, Any> = HashMap()
+        replyData["text"] = text
+//        TODO check object nullability
+        bool?.let { replyData["boolean"] = bool }
+        replyData["number"] = number
+        replyData["checklist"] = checklist
+        val reply = createReplyWithData(replyData)
+        return api.createReply(metaformId, reply, null, ReplyMode.REVISION.toString())
+    }
+
+    /**
+     * Creates new reply for the simple form
+     *
+     * @param metaform  metaform
+     * @param value     value
+     * @param replyMode reply model
+     * @return reply
+     */
+    @Throws(IOException::class)
+    fun createSimpleReply(metaformId: UUID, value: String, replyMode: ReplyMode): Reply {
+        val replyData1: MutableMap<String, Any> = HashMap()
+        replyData1["text"] = value
+        val reply = createReplyWithData(replyData1)
+        return create(metaformId, replyMode.toString(), reply)
+    }
+
+    /**
+     * Asserts reply can not be found with given owner key
+     *
+     * @param metaformId metaform id
+     * @param reply    reply
+     * @param ownerKey owner key
+     */
+    @Throws(IOException::class)
+    fun assertReplyOwnerKeyFindForbidden(metaformId: UUID, replyId: UUID, ownerKey: String?) {
+        try {
+            api.findReply(metaformId, replyId, ownerKey)
+            Assert.fail(String.format("Should not be able to find reply %s", replyId.toString()))
+        } catch (e: ClientException) {
+            Assert.assertEquals(403, e.statusCode.toLong())
+        }
+    }
+
+    /**
+     * Finds a reply
+     *
+     * @param metaformId metaform id
+     * @param replyId    reply id
+     * @param ownerKey   owner key
+     * @return found reply
+     */
+    @Throws(IOException::class)
+    fun findReply(metaformId: UUID, replyId: UUID, ownerKey: String?): Reply {
+        return api.findReply(metaformId, replyId, ownerKey)
+    }
+
+    /**
+     * Updates a reply into the API
+     *
+     * @param metaformId metaform id
+     * @param body       body payload
+     * @param ownerKey   owner key
+     */
+    @Throws(IOException::class)
+    fun updateReply(metaformId: UUID, replyId: UUID, body: Reply, ownerKey: String?) {
+        api.updateReply(metaformId, replyId, body, ownerKey)
+    }
+
+    /**
+     * Deletes a reply from the API
+     *
+     * @param metaformId metaform id
+     * @param reply      reply to be deleted
```

Comment:
```suggestion
     * @param replyId      reply to be deleted
```
Was documented wrong